### DISCUSSION TITLE
Frequency of data updates

### DISCUSSION THREAD

**Discussion by Feggla** [cwendt94/espn-api#508] [ESPN API]
I've been using this package for a few years now for some personal projects and really appreciate the work you've put into it. Just wondering about api update frequency - at the moment the data I get updates daily at around 6am ET. Is there any way to update more frequently using this package, specifically for NBA fantasy stats? 

Side-question: Played around with the api a lot and can't find a way to get daily NBA player stats - is this currently possible?

**Comment by cwendt94** [comment]
The box score info should be updated real time but you would need to setup some type of re fetch logic on a certain interval to get those updates.

You can get a players daily stats by using `player_info` call. That will return the stats of each game they played.

**Comment by Feggla** [comment]
The box_score method doesn't really return the data I'm after, I get all of my stats from the scoreboard method, but I'm unable to get live stats for either. I'll post my a quick example below, appreciate you taking a look:

`league = League(league_id=xxx, year=2024, espn_s2='xxx', swid='xxx')
box_scores = league.scoreboard(matchupPeriod=7)

for matchups in box_scores:
    print(matchups.home_team, matchups.home_team_cats)`

Output:
`Team(xxx) {'PTS': {'score': 2.0, 'result': 'LOSS'}, 'BLK': {'score': 1.0, 'result': 'LOSS'}, 'STL': {'score': 0.0, 'result': 'LOSS'}, 'AST': {'score': 4.0, 'result': 'LOSS'}, 'REB': {'score': 7.0, 'result': 'TIE'}, '38': {'score': 0.0, 'result': 'TIE'}, 'TO': {'score': 2.0, 'result': 'TIE'}, 'FGM': {'score': 1.0, 'result': None}, 'FGA': {'score': 3.0, 'result': None}, 'FTM': {'score': 0.0, 'result': None}, 'FTA': {'score': 0.0, 'result': None}, '3PTM': {'score': 0.0, 'result': 'LOSS'}, '3PTA': {'score': 1.0, 'result': None}, 'FG%': {'score': 0.33333333, 'result': 'LOSS'}, 'FT%': {'score': 0.0, 'result': 'LOSS'}, '3PT%': {'score': 0.0, 'result': 'LOSS'}}`

This output has the category data from the previous day, and does not include any of the data from today's games, which concluded a few hours ago. The box_scores method also returns the matchup score prior to todays games.

**Reply by cwendt94** [reply]
If you pass matchup_total = false it should give the live data for that day. So it should be something like `box_scores = league.box_scores(matchup_period=7, matchup_total=False)`

**Reply by Feggla** [reply]
Will give that a go for box_scores - is there an equivalent for the 'scoreboard' method, which is where I get my category data from?

**Reply by cwendt94** [reply]
Box scores should also show all of your category data, here is the info on the fields https://github.com/cwendt94/espn-api/wiki/Box-Score-Class-Basketball

**Reply by Feggla** [reply]
Have just tried the below, approximately an hour into the current matchday - unfortunately the scoreboard stats still returns data from the previous day, and doesn't include any current matchday info. 

box_scores = league.box_scores(matchup_period=7, matchup_total=False)

for matchups in box_scores:
    print(matchups.away_team, matchups.away_stats)

**Reply by cwendt94** [reply]
Hmm you might have to also pass in the scoring_period as well. I will try to play around with my league once a game starts to see if it will return updated data.

**Reply by Feggla** [reply]
Any luck with this one? I'll have another try with scoring_period in the next couple of days while matches are live.

**Comment by Feggla** [comment]
I've had no luck getting up to date stats from the box scores - any further ideas?

**Reply by cwendt94** [reply]
Sorry, I have been busy the last couple weeks before the holidays. I haven't been able to play around with it but hopefully will have time in the next week to see if I can find a way to get live updates.

**Comment by Feggla** [comment]
Have come back around to this and still no luck getting up-to-date info when pulling either the Scoreboard or Boxscores classes. Am I right in assuming it might not be possible to pull the current info?

**Comment by Feggla** [comment]
Running into the same issue for 24/25 leagues again unfortunately - I've looked through the base code but nothing jumping out as to why updates are only occurring once daily. Any chance anyone has an idea on this?

**Reply by cwendt94** [reply]
I am currently checking the box_score stats for a live game right now and seeing points being updated when I refresh the League.

**Reply by Feggla** [reply]
Really appreciate you taking the time to look into this one!

Any chance you can provide some example code for your calls? I've just run the below and am not getting updated stats from the current games, only from end of play yesterday:

scoreboard = league.box_scores(matchup_period=1, scoring_period=1, matchup_total=False)

I've tried using a range of scoring_period=1-4, and changed matchup_total to True, can't seem to load current data using the .home_team_cats or .away_team_cats.

**Reply by cwendt94** [reply]
I haven't checked the CAT stats, I was just checking the player stats
```python3
>>> league = League(501585302, 2025)
>>> box = league.box_scores(matchup_total=False)
>>> box[0].away_lineup[0]
Player(Luka Doncic, points:0)
>>> box[0].away_lineup[0].stats
{'3': {'applied_total': 0, 'applied_avg': 0, 'team': 'SAS', 'date': datetime.datetime(2024, 10, 24, 23, 30), 'avg': None, 'total': {'PTS': 3.0, 'BLK': 0.0, 'STL': 0.0, 'AST': 2.0, 'OREB': 0.0, 'DREB': 3.0, 'REB': 3.0, 'EJ': 0.0, 'FF': 0.0, 'PF': 0.0, 'TF': 0.0, 'TO': 1.0, 'DQ': 0.0, 'FGM': 1.0, 'FGA': 5.0, 'FTM': 1.0, 'FTA': 2.0, '3PM': 0.0, '3PA': 2.0, 'FG%': 0.2, 'FT%': 0.5, '3PT%': 0.0, 'AFG%': 0.2, 'FGMI': 4.0, 'FTMI': 1.0, '3PMI': 2.0, 'APG': 2.0, 'BPG': 0.0, 'MPG': 8.0, 'PPG': 3.0, 'RPG': 3.0, 'SPG': 0.0, 'TOPG': 1.0, '3PG': 0.0, 'PPM': 0.375, 'A/TO': 2.0, 'STR': 0.0, 'DD': 0.0, 'TD': 0.0, 'QD': 0.0, 'MIN': 8.0, 'GS': 1.0, 'GP': 1.0, 'TW': 0.0, 'FTR': 0.4}}}
```

**Reply by Feggla** [reply]
Ahh right - I've got my setup configured to pull data from the team cats for h2h category leagues with the .home_team_cats. Any idea why that would be different/where the home_team_cats data is aggregated? Is it directly from ESPN or do you aggregate it from the player stats in this package?

**Reply by cwendt94** [reply]
where are you pulling `home_team_cats ` from? I don't see that in box_score

**Reply by Feggla** [reply]
Ahh really sorry for the confusion, realised I'm using a combination of league.box_scores and league.scoreboard with my setup. 

score = league.scoreboard(matchupPeriod=matchup_period)
score[0].home_team_cats is the data I'm reading, which is updated daily. Any possibility of live stats with this scoreboard method?

**Reply by cwendt94** [reply]
Ahh okay that makes sense, I know box_score will definitely get the live stats. I will have to look into scoreboard API call a little and see how its being grabbed!

**Reply by cwendt94** [reply]
From a quick look at the API and the data ESPN returns I think for matchup it will only be after the end of games for the day. I think the only way to get some live information is through box_score

**Reply by Feggla** [reply]
Any idea where that gets determined? I'd love to take a look and see if I can do some work on getting live scoreboard stats, as using the box_score would be pretty tough for my use case.

**Reply by Feggla** [reply]
You know what, box_score does have a .home_stats and .away_stats attribute (which has the category data structures similarly to scoreboard.home_team_cats), but same issue with not updating with current stats.

**Reply by cwendt94** [reply]
Interesting yeah I would think it can get the latest as you see it being updated on ESPN's website. If you want to look into it a little more that would be great! This is the API call that is getting the information from

https://lm-api-reads.fantasy.espn.com/apis/v3/games/fba/seasons/2025/segments/0/leagues/501585302?view=modular&view=mNav&view=mMatchupScore&view=mScoreboard&view=mSettings&view=mTopPerformers&view=mTeam

This is a public league so you should be able to see all of the data. Scoreboard and Box Score grab from the `schedule` field.
